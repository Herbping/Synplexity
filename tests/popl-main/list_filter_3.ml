type nat =
  | Z
  | S of nat

type list =
  | Nil
  | Cons of nat * list

type bool =
  | True
  | False

let rec list_filter_3 :list -> (nat -> bool) -> list |>
/\([] -> /\(0 -> True, 1 -> False, 2 -> True) -> [],
   [0] -> /\(0 -> True, 1 -> False, 2 -> True) -> [0],
   [1] -> /\(0 -> True, 1 -> False, 2 -> True) -> [],
   [2] -> /\(0 -> True, 1 -> False, 2 -> True) -> [2],
   [0; 0] -> /\(0 -> True, 1 -> False, 2 -> True) -> [0; 0],
   [0; 1] -> /\(0 -> True, 1 -> False, 2 -> True) -> [0],
   [] -> /\(0 -> False, 1 -> True, 2 -> True) -> [],
   [1] -> /\(0 -> False, 1 -> True, 2 -> True) -> [1],
   [0; 1] -> /\(0 -> False, 1 -> True, 2 -> True) -> [1],
   [0; 0; 1] -> /\(0 -> False, 1 -> True, 2 -> True) -> [1]) = ?
