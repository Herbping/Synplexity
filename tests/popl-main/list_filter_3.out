(* Generated Thu Oct  8 20:41:00 2015 *)
fix f1 (l1:list) : (nat -> bool) -> list ->
  fix f2 (f3:nat -> bool) : list ->
    match l1 with
    | Nil u1 -> Nil
    | Cons t1 -> match f3 (#1 t1) with
      | True u1 -> Cons (#1 t1, (f1 (#2 t1)) f3)
      | False u1 -> (f1 (#2 t1)) f3
